src/main/java/com/hbm/items/tool/ItemPipette.java

```diff
@@ -0,0 +1,175 @@
+package com.hbm.items.tool;
+
+import api.hbm.fluid.IFillableItem;
+
+import com.hbm.inventory.fluid.FluidType;
+import com.hbm.inventory.fluid.Fluids;
+import com.hbm.items.ModItems;
+
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+import net.minecraft.client.renderer.texture.IIconRegister;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.ChatComponentText;
+import net.minecraft.util.IIcon;
+import net.minecraft.world.World;
+
+import java.util.List;
+
+public class ItemPipette extends Item implements IFillableItem {
+
+    public ItemPipette() {
+        this.setMaxDamage(0);
+    }
+
+    @SideOnly(Side.CLIENT) protected IIcon overlayIcon;
+
+    public int amount = this.getMaxFill();
+
+    public FluidType type = Fluids.NONE;
+
+    public int getMaxFill() {
+        if(this == ModItems.pipette_laboratory)
+            return 50;
+        else
+            return 1_000;
+    }
+
+    public void initNBT(ItemStack stack) {
+
+        stack.stackTagCompound = new NBTTagCompound();
+
+        this.setFill(stack, type, 0);
+    }
+
+    public void setFill(ItemStack stack, FluidType type, int fill) {
+        if(!stack.hasTagCompound()) {
+            initNBT(stack);
+        }
+
+        this.type = type;
+        stack.stackTagCompound.setInteger(type.getName(), fill);
+    }
+
+    public int getFill(ItemStack stack, FluidType type) {
+        if(!stack.hasTagCompound()) {
+            initNBT(stack);
+        }
+
+        return stack.stackTagCompound.getInteger(type.getName());
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public ItemStack onItemRightClick(ItemStack stack, World world, EntityPlayer player) {
+
+        if(!stack.hasTagCompound()) {
+            initNBT(stack);
+        }
+
+        if(!world.isRemote) {
+            if (this.getFill(stack, type) == 0) {
+
+                if(this != ModItems.pipette_laboratory)
+                    this.amount = player.isSneaking() ? Math.min(this.amount + 50, 1_000) : Math.max(this.amount - 50, 50);
```

Comment:
i genuinely can't even tell what this code is supposed to do, or why you're handling the amount at all when you just checked the amount to be 0

Reply:
It's meant to allow the player to change the amount of fluid in the pipette, probably gonna rename the variable from `amount` to `maxAmount` or something of the like. It checks if the fill is 0, and if so allows the changing of the max capacity in units of either 1mB or 50mB, depending on if it's a laboratory pipette or a normal one.